### ISSUE TITLE
VirtualBox Guest Addition xzm builder/process not available

### ISSUE THREAD

**Issue by CemHyur** [porteux/porteux#5]
Please help with VirtualBox Guest Addition xzm...

There is no VirtualBox Guest Addition Builder to build Guest Addition xzm module, nor any script or something like to build.

Or provide any process/ script for it (I'm newbie to Linux & github) (Windows user)

Please help to create xzm module of Guest Addition...
Installation of Guest Addition is working fine...

Thank You...

**Comment by porteux** [comment]
Done. You can download it in the app store.

Thanks for the feedback. :)

**Comment by porteux** [comment]
I've just tested and it works if the different VMs are using the same kernel as the machine where the Guest Additions module was built. It works even from stable to current and vice-versa. So probably what's happening in your case is that these VMs you're using have different kernels -- you can check it by running `check-kernel` in the terminal. Also, make sure you're using the same VirtualBox version and it matches the Guest Additions version.

And yes, you can generate a VirtualBox Guest Additions module from any version you want. First you need to download the Guest Additions ISO with the version you want (https://download.oracle.com/virtualbox), then make sure you have updated  PorteuX App Store to latest version (just run it once and it will autoupdate). Now go to the terminal and run: `sudo /opt/porteux-scripts/porteux-app-store/virtualbox-guestadditions-builder.sh [VBoxGuestAdditions_xxxx.iso]` where `[VBoxGuestAdditions_xxxx.iso]` parameter is the path to the .iso you have downloaded.

Let me know if this works for you.
Cheers

**Comment by CemHyur** [comment]
Ok...
Sorry it is working...

Actually it was working but when i generated xzm it's automatically mounted vboxsf, and in other pc wasn't...
so I mount vboxsf via terminal...

When I didn't get vboxsf mounted, so i though it's not working...

It's working...

Thank you...

**Comment by CemHyur** [comment]
Login=
Cheatcode not working...How to use in correct way???

I have added cheatcode "login=" in APPEND line in porteux.cfg at /boot/syslinux/porteux.cfg in Vbox Harddisk... After rebooting its auto login into guest account...

Also try login=root still login auto into guest...

**Comment by porteux** [comment]
Yes, this is a known issue. It's already fixed in 0.3 and it will be released soon.

Thanks for reporting! :)

**Comment by CemHyur** [comment]
Thanks :)

and as root login also not working,
tried via cheatcode and
tried via login screen (after logout from guest)

in porteux 0.2 lxqt stable...
.
.
.
Hope porteux 0.3 release soon :)

**Comment by CemHyur** [comment]
```
Hi
I try as you suggest, but no module generated in /tmp nor in /optional

> go to the terminal and run: sudo /opt/porteux-scripts/porteux-app-store/virtualbox-guestadditions-builder.sh [VBoxGuestAdditions_xxxx.iso]


here is I got output:

guest@porteux:~$ sudo /opt/porteux-scripts/porteux-app-store/virtualbox-guestadditions-builder.sh /mnt/sda1/porteux/optional/VBoxGuestAdditions_7.0.6.iso 
mount: /mnt/virtualbox-guestadditions: WARNING: source write-protected, mounted read-only.
Verifying archive integrity...  100%   MD5 checksums are OK. All good.
Uncompressing VirtualBox 7.0.6 Guest Additions for Linux  100%  
VirtualBox Guest Additions installer
VirtualBox Guest Additions: Starting.
VirtualBox Guest Additions: Setting up modules
VirtualBox Guest Additions: Building the VirtualBox Guest Additions kernel modules.  This may take a while.
VirtualBox Guest Additions: To build modules for other installed kernels, run
VirtualBox Guest Additions:   /sbin/rcvboxadd quicksetup <version>
VirtualBox Guest Additions: or
VirtualBox Guest Additions:   /sbin/rcvboxadd quicksetup all
VirtualBox Guest Additions: Building the modules for kernel 6.2.11-porteux.
VirtualBox Guest Additions: Running kernel modules will not be replaced until the system is restarted Setting up modules
VirtualBox Guest Additions kernel modules 7.0.6 r155176 are already available for kernel 6.2.11-porteux and do not require to be rebuilt.
Running kernel modules will not be replaced until the system is restarted
vboxadd-service.sh: Starting VirtualBox Guest Addition service.
Destination  is not writable. New module placed in /tmp and not activated.
guest@porteux:~$


Sorry for the trouble.
